### PR TITLE
WFLY-14953 distributable-ejb subsystem

### PR DESCRIPTION
**Pull request by rachmatowicz**
This PR adds the distrbutable-ejb subsystem to wildfly, as described in https://issues.redhat.com/browse/WFLY-14953

### CODE REVIEW

**Review by fabiobrz — COMMENTED** [review]
Thanks @rachmatowicz - I had a quick look at the test development scope in here and it seems to me that it reflects what defined by the feature analysis document. The same goes for the community docs part.

**Comment by pferraro** [comment]
Added hold label temporarily, while @rachmatowicz works out squashing commits.

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3Extension.java [hunk 1]
```diff
@@ -64,10 +64,11 @@
     public static final String NAMESPACE_7_0 = EJB3SubsystemNamespace.EJB3_7_0.getUriString();
     public static final String NAMESPACE_8_0 = EJB3SubsystemNamespace.EJB3_8_0.getUriString();
     public static final String NAMESPACE_9_0 = EJB3SubsystemNamespace.EJB3_9_0.getUriString();
+    public static final String NAMESPACE_10_0 = EJB3SubsystemNamespace.EJB3_10_0.getUriString();
 
     static final PathElement SUBSYSTEM_PATH = PathElement.pathElement(ModelDescriptionConstants.SUBSYSTEM, SUBSYSTEM_NAME);
 
-    static final ModelVersion CURRENT_MODEL_VERSION = EJB3Model.VERSION_9_0_0.getVersion();
+    static final ModelVersion CURRENT_MODEL_VERSION = EJB3Model.VERSION_10_0_0.getVersion();
```
Comment (pferraro) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3Extension.java:71]: This constant duplicates EJB3Model.CURRENT.  I suggest removing this constant and replacing any references with EJB3Model.CURRENT.getVersion().

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3RemoteResourceDefinition.java [hunk 2]
```diff
@@ -78,6 +79,7 @@
                     .setDefaultValue(new ModelNode(BeanManagerFactoryServiceConfiguratorConfiguration.DEFAULT_CONTAINER_NAME))
                     // TODO: replace this with a Requirement reference when the ejb-spi module for clustering is available
                     .setCapabilityReference(INFINISPAN_CACHE_CONTAINER_CAPABILITY_NAME, EJB_REMOTE_CAPABILITY)
+                    .setDeprecated(ModelVersion.create(10,0,0))
```
Comment (pferraro) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3RemoteResourceDefinition.java:82]:
```java
.setDeprecated(EJB3Model.VERSION_10_0_0.getVersion())

**Review by pferraro — CHANGES_REQUESTED** [review]

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3Extension.java (hunk 1 above)
Comment (rachmatowicz) [reply ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3Extension.java:71]: Done.

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3RemoteResourceDefinition.java (hunk 2 above)
Comment (rachmatowicz) [reply ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3RemoteResourceDefinition.java:82]: Done.

**Review by pferraro — APPROVED** [review]

**Pull request merged**